pool/entities.go

```diff
@@ -0,0 +1,170 @@
+package pool
+
+type ID string
+
+type Name string
+
+// I don't know i have to bring Peer Available storage and other metric inside Peer
+type Peer string
+
+type ReplicationFactor int8
+
+// Requirement of joining the pool
+type Requirement struct {
+	Latency   string `json:"latency"`
+	Bandwidth string `json:"bandwidth"`
+	ReplicationFactor ReplicationFactor `json:"rep_factor"`
+}
+
+// ResourceRequest object (it is much more complicated for simplicity we think as file).
+type Resource struct {
+	ID  ID `json:"id"`
+	FileName  Name
+	Owner     Peer
+	Size      string
+	ReplicationFactor ReplicationFactor
+	Allocation []Peer
+}
+
+
+// Event is a domain event marker.
+type Event interface {
```

Comment:
any reason Event is not a struct? this is the standard?

Reply:
:))